### PR TITLE
Sc 22733 ruby 3.2 upgrade

### PR DESCRIPTION
**Pull request by Sparkmasterflex**
* Fixed deprecation for upgrade to ruby 3.2
* Opentransact gem is crazy old and abandoned
    * it doesn't look like we're using this in LM so I'm commenting it out and removing the gem from here
* Fixed tests due to rspec upgrade

IDK if we can/should disconnect this from the repo we forked it from but I believe there's a bunch of stuff in this gem that we're not using and unless we're using it in other RG projects we can probably just move to LM codebase

### CODE REVIEW

File: lib/oauth/models/consumers/services/opentransact_token.rb
Comment (aaronvb) [inline lib/oauth/models/consumers/services/opentransact_token.rb:1]: Is it okay to just remove all code that is commented out or this whole file?

**Review by aaronvb — APPROVED** [review]
LGTM, one non-blocking question.

> IDK if we can/should disconnect this from the repo we forked it from but I believe there's a bunch of stuff in this gem that we're not using and unless we're using it in other RG projects we can probably just move to LM codebase

I don't think we can disconnect it from the fork unless we remade the gem from scratch. It's worth noting that this is an open source project and we forked it from an open source project so maybe someone else can find this useful? If not then I guess we could create our own internal gem since it is an MIT license.

Also github packages! 😝

File: lib/oauth/models/consumers/services/opentransact_token.rb
Comment (Sparkmasterflex) [reply lib/oauth/models/consumers/services/opentransact_token.rb:1]: I don't know yet. I'm waiting to get it published within our RGs gems realm and fully test it first

**Comment by Sparkmasterflex** [comment]
I would say we could try and get a PR for the main repo but it looks like the original has been abandoned and it's using that opentransact gem that is even more abandoned. I doubt it's worth the trouble.

It might be worth creating a new open source gem with just the parts that we're specifically using but also don't know how useful that'll be to the rest of the community. It really seems like this is just a wrapper gem 🤷

**Comment by aaronvb** [comment]
> I would say we could try and get a PR for the main repo but it looks like the original has been abandoned and it's using that opentransact gem that is even more abandoned. I doubt it's worth the trouble.
> 
> It might be worth creating a new open source gem with just the parts that we're specifically using but also don't know how useful that'll be to the rest of the community. It really seems like this is just a wrapper gem 🤷

@Sparkmasterflex Is there anything wrong or have concerns about keeping this as a fork?

**Comment by Sparkmasterflex** [comment]
Mostly just the added mental overhead and the feeling of less ownership on it. Honestly we almost never even think about this thing so IDK if it's worth doing anything now but when we go to do more upgrades we're just going to run into this situation again

**Comment by aaronvb** [comment]
> Mostly just the added mental overhead and the feeling of less ownership on it. Honestly we almost never even think about this thing so IDK if it's worth doing anything now but when we go to do more upgrades we're just going to run into this situation again

Makes sense. We should be able to clone and push this to a new private repo while keeping all the history intact.

**Pull request merged**